pkg/manager/member/tidb_failover.go

```diff
@@ -35,6 +36,17 @@ func (tf *tidbFailover) Failover(tc *v1alpha1.TidbCluster) error {
 		tc.Status.TiDB.FailureMembers = map[string]v1alpha1.TiDBFailureMember{}
 	}
 
+	for _, tidbMember := range tc.Status.TiDB.Members {
```

Comment:
These logics are recoveries, not failover. Recovery is here: https://github.com/weekface/tidb-operator/blob/115fe19f348e8546f4e10705e4729fd0aa9236d0/pkg/manager/member/tidb_member_manager.go#L150-L149

Reply:
this logic is hard to add to recovery function, cause recovery currently only happen when all members are running.